src/engine/enginebufferscale.h

```diff
@@ -57,24 +56,25 @@ class EngineBufferScale : public QObject {
     //
     // If parameter settings are outside of acceptable limits, each setting will
     // be set to the value it was clamped to.
-    // TODO: move samplerate changes to a separate function since that's often
-    // an unrelated change and doesn't happen often.
-    virtual void setScaleParameters(int iSampleRate,
-                                    double base_rate,
+    virtual void setScaleParameters(double base_rate,
                                     double* pTempoRatio,
                                     double* pPitchRatio) {
-        m_iSampleRate = iSampleRate;
         m_dBaseRate = base_rate;
         m_dTempo = *pTempoRatio;
         m_dPitch = *pPitchRatio;
     }
 
-    /** Get new playpos after call to scale() */
+    // Set the desired output sample rate.
+    virtual void setSampleRate(int iSampleRate) {
+        m_iSampleRate = iSampleRate;
+    }
+
+    // Get new playpos after call to scale()
     double getSamplesRead();
-    /** Called from EngineBuffer when seeking, to ensure the buffers are flushed */
+    // Called from EngineBuffer when seeking, to ensure the buffers are flushed */
     virtual void clear() = 0;
-    /** Scale buffer */
-    virtual CSAMPLE* getScaled(unsigned long buf_size) = 0;
+    // Scale buffer
+    virtual void getScaled(CSAMPLE* pOutput, const int iBufferSize) = 0;
```

Comment:
this should return the number of samples read, so we can return a value != iBufferSize in case of error.